src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -28,19 +36,29 @@ public Movie addMovie(Movie movie) {
     @Override
     public List<Movie> getAllMovies() {
         List<Movie> movieData = readJsonData();
+        if (movieData.isEmpty()) {
+            throw new MovieNotFoundException("no movies found");
+        }
         return movieData;
     }
 
-
     @Override
     public Movie getMovie(String movieId) {
         List<Movie> movieData = readJsonData();
-        return movieData.parallelStream().filter(mov -> mov.getId().equals(movieId)).findAny().orElse(null);
+        Movie searchedMovie = movieData.parallelStream().filter(mov -> mov.getId().equals(movieId)).findAny().orElse(null);

```

Comment:
**findAny** will return **Optional**. It's better to use it  (if(optionalObject.isEmpty()) {///exception}) then use **.orElse(null)** and then check if not null.